smartsim/_core/entrypoints/telemetrymonitor.py

```diff
@@ -0,0 +1,688 @@
+    if parent_keys:
+        container = "shards" if "shards" in parent_keys else "models"
+        child_type = "orchestrator" if container == "shards" else "model"
+        for child_entity in persistable_entity[container]:
+            entity = _hydrate_persistable(child_entity, child_type, str(exp_dir))
+            entities.append(entity)
+
+        return entities
+
+    entity = _hydrate_persistable(persistable_entity, entity_type, str(exp_dir))
+    entities.append(entity)
+    return entities
+
+
+def hydrate_persistables(
+    entity_type: str,
+    run: t.Dict[str, t.Any],
+    exp_dir: pathlib.Path,
+) -> t.Dict[str, t.List[JobEntity]]:
+    """Map a collection of entity data persisted in a manifest file to an object"""
+    persisted: t.Dict[str, t.List[JobEntity]] = {
+        "model": [],
+        "orchestrator": [],
+        "ensemble": [],
+    }
+    for item in run[entity_type]:
+        entities = hydrate_persistable(entity_type, item, exp_dir)
+        for new_entity in entities:
+            persisted[new_entity.type].append(new_entity)
+
+    return persisted
+
+
+def hydrate_runs(
+    persisted_runs: t.List[t.Dict[str, t.Any]], exp_dir: pathlib.Path
+) -> t.List[Run]:
+    """Map run data persisted in a manifest file to an object"""
+    the_runs: t.List[Run] = []
+    for run_instance in persisted_runs:
+        run_entities: t.Dict[str, t.List[JobEntity]] = {
+            "model": [],
+            "orchestrator": [],
+            "ensemble": [],
+        }
+
+        for key in ["model", "orchestrator", "ensemble"]:
+            _entities = hydrate_persistables(key, run_instance, exp_dir)
+            for entity_type, new_entities in _entities.items():
+                if new_entities:
+                    run_entities[entity_type].extend(new_entities)
+
+        run = Run(
+            run_instance["timestamp"],
+            run_entities["model"],
+            run_entities["orchestrator"],
+            run_entities["ensemble"],
+        )
+        the_runs.append(run)
+
+    return the_runs
+
+
+def load_manifest(file_path: str) -> t.Optional[RuntimeManifest]:
+    """Load a persisted manifest and return the content"""
+    manifest_dict: t.Optional[t.Dict[str, t.Any]] = None
+    try_count = 1
+
+    while manifest_dict is None and try_count < 6:
+        source = pathlib.Path(file_path)
+        source = source.resolve()
+
+        try:
+            if text := source.read_text(encoding="utf-8").strip():
+                manifest_dict = json.loads(text)
+        except json.JSONDecodeError as ex:
+            print(f"Error loading manifest: {ex}")
+            # hack/fix: handle issues reading file before it is fully written
+            time.sleep(0.5 * try_count)
+        finally:
+            try_count += 1
+
+    if not manifest_dict:
+        return None
+
+    exp = manifest_dict.get("experiment", None)
+    if not exp:
+        raise ValueError("Manifest missing required experiment")
```

Comment:
Not wrong just curious: any reason the first invalid state returns a null-ish value while the second one raises an exception?

Reply:
You can try to load an empty file or maybe a file that doesn't exist. That's expected behavior (return None)

I figured that if I load a file and it has content that is not in the right format, I cannot recover so I throw.